app/controllers/chargeback_controller.rb

```diff
@@ -273,9 +300,136 @@ def cb_rate_form_field_changed
     cb_rate_get_form_vars
   end
 
+  # AJAX driven routine to storage the selected rate details for remove
+  def cb_selected_rd
+    @edit = session[:edit]
+    unless params[:select_all].nil?
```

Comment:
Is it possible for params[:select_all] and params[:select] to be non-nil?

They seem to be completely independent things... maybe extract them to separate methods based on the select/select_all value?  

```ruby
if params[:select]
  process_select(any_local_variables_you_need)
elsif params[:select_all]
  process_select_all(any_local_variables_you_need)
end
```

You might find that they're doing similar things and you can further refactor them once they're in methods.